hugo/layouts/_default/list.algolia.json

```diff
@@ -18,8 +18,12 @@
     {{ $cutoff := 1000 -}}
     {{ $index := .index -}}
     {{ $skipWords := mul $cutoff $index }}
+    {{ $fileID := delimit (shuffle (split (md5 $page.Title) "" )) "" }} {{/* generates a "random" string based on the page title */}}
```

Comment:
maybe movie this to line 23-24 in the else statement because reading it now looks a bit like the random id is the default (which we do now want of course because that would result in a lot of duplicate entries in algolia)

Reply:
another note: don't use shuffle on the title because that would result in a different id with each render. Algolia uses objectId to check if an entry exists or not and generating random as default will cause duplicates in algolia because the id changes

Reply:
What is the uniqueness we are trying to go for here? Unique per page? Or unique per page per commit?

Reply:
in 99.99% of the cases we do have file-Id which is unique but we have to check for this because it can be empty and right now Hugo gives a warning on build for this line. That's why we had to have a fallback

Reply:
In what cases can the file-id be empty? 

Is there anything we can/should rely on as far as the preprocessor is concerned?

Reply:
the warning in the build was:
debug: hugo: WARN  .File.UniqueID on zero object. Wrap it in if or with: {{ with .File }}{{ .UniqueID }}{{ end }}
@anne-usmedia Did you find out which page triggered this error?

Reply:
The tags (/tags/language) pages are the ones that do not have a .File, and therefore wouldn't have an objectID in the algolia.json.

Reply:
I've created a new issue for the tag-overview pages which should not exist: https://linear.app/usmedia/issue/CUE-310
But this is already happening in current alpha so isn't caused by the update to hugo 0.117.
We still should have a fallback for non file pages since hugo gives warnings about them now but we have to make sure they don't change on build because they need to be the same as algolia uses them to identify the item. So we can't use randomize or loop index because these change Maybe use slugified page title or something as a fallback for automaticly genertated pages? Even if we do not use them we do need to have something in place because we want to get rid of the current warning on build.